scripts/market_parameters.ipynb

```diff
@@ -179,89 +282,321 @@
+    "            trades=trades,\n",
+    "            L0=L0,\n",
+    "            L1=L1,\n",
+    "            market_fee=market_fee,\n",
+    "            mech_cost=mech_cost,\n",
+    "            mech_calls_per_trade=mech_calls_per_trade,\n",
+    "            win_option=win_option)\n",
+    "\n",
+    "        # The market probability of an option is proportional to its liquidity\n",
+    "        probability = L0_result / (L0_result + L1_result)\n",
+    "        p0_total += probability\n",
+    "        #print(earnings)\n",
+    "        total_final_amount += sum(final_amounts)\n",
+    "\n",
+    "    avg_probability = p0_total / iterations\n",
+    "\n",
+    "    if nTrades0 + nTrades1 > 0:\n",
+    "        avg_roi = (total_final_amount - total_initial_amount) / total_initial_amount\n",
+    "    else:\n",
+    "        avg_roi = 0\n",
+    "\n",
+    "    return np.array([avg_probability, avg_roi])\n",
+    "\n",
+    "\n",
+    "def tests() -> None:\n",
+    "    # Test trade all\n",
+    "    trades = [(1.,0), (1.,0), (0.8,0), (0.8,0), (0.8,0), (0.32,0)]\n",
+    "    L0 = L1 = 10\n",
+    "    (L0, L1, earnings) = fpmm_trade_all(trades, L0, L1, 0.02, 0)\n",
+    "    p = L0/(L0+L1)\n",
+    "    expected_value = 0.6814355029609638\n",
+    "    tolerance = 1e-6\n",
+    "    assert abs(expected_value - p) < tolerance, \"Market simulation does not match expected result.\"\n",
+    "    \n",
+    "    # Test simulations\n",
+    "    assert abs(simulate(4, 0, 1.0, 10, 10, 0.02, 0, 1, 0, 1)[1] - 0.6840229885057476) < tolerance, \"Market simulation does not match expected result: 4 win trades\"\n",
+    "    assert simulate(0, 4, 1.0, 10, 10, 0.02, 2, 1, 0, 1)[1] == 0, \"Market simulation does not match expected result: ROI for mech costs > available amount != 0. \"\n",
+    "    assert simulate(0, 4, 1.0, 10, 10, 0.02, 0, 1, 0, 1)[1] == -1, \"Market simulation does not match expected result: ROI for all trades lose != -1.\"\n",
+    "\n",
+    "\n",
+    "def tests2() -> None:\n",
+    "    L0 = 11000000000000000000\n",
+    "    L1 = 9000000000000000000\n",
+    "    trades = [\n",
+    "        (1199999999999994880,1),\n",
+    "        (210820759921632064,1),\n",
+    "        (469987407320034624,1),\n",
+    "        (731356169902985344,1)\n",
+    "    ]\n",
+    "    (L0, L1, earnings) = fpmm_trade_all(trades, L0, L1, 0.02, 0)\n",
+    "    print(L0)\n",
+    "    print(L1)\n",
+    "    print(\"\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=1199999999999994880, option=1, L0=11000000000000000000, L1=9000000000000000000, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"9728773584905664512, 10175999999999994880\")\n",
+    "    print(\"Ok\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=210820759921632064,  option=1, L0=9728773584905664512,  L1=10175999999999994880, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
```

Comment:
This is wrong. This trade has the mech output with p_yes > p_no so the voted option in this trade is option=0